UG + DG update - overall update (#166)
AY2223S2-CS2103-W17-1/tp: docs/UserGuide.md

* `add n/John Doe p/98765432 e/johnd@example.com a/John street, block 123, #01-01 c/948372948`
* `add n/Betsy Crowe t/primary c/83927482 e/betsycrowe@example.com a/Downtown p/1234567 t/primary3`

<div markdown="span" class="alert alert-warning">:exclamation:
Caution: </div> 

> * In order to avoid potential confusion, we do not allow any student to have the same name regardless of the letter cases (e.g. Harry, HARRY, harry are deemed as same name). If you wish to add in a student who has the same name as an existing student in the student list, consider append a index number after the student's name to distinguish them.

Reviewer: ```suggestion
* In order to avoid potential confusion, we do not allow any student to have the same name regardless of the letter cases (e.g. Harry, HARRY, harry are deemed as same name). If you wish to add in a student who has the same name as an existing student in the student list, consider adding a number after the student's name to distinguish them.
```

Author: Noted! Thank you

  e.g. Emily and Emily 2 are acceptable names.

> * We allow `PHONE_NUMBER` and `CONTACT_PARENT` to be the same as not all student have their own phone.

Reviewer: I think this can be moved to the 'Tip' sections above instead as I feel like it isn't really a caution (Can change according to my PR #169 and use that instead) 😄

Author: Sure! Thank you for your suggestion.

* Edits the student at the specified `INDEX`. The index refers to the index number shown in the displayed student list. The index **must be a positive integer** 1, 2, 3, …​
* At least one of the optional fields must be provided.
* Existing values will be updated to the input values.
* When editing tags, all the existing tags of the student will be removed. i.e. adding of tags is not cumulative.
* You can remove all the student’s tags by typing `t/` without specifying any tags after it.
* We allow you to edit the `PHONE_NUMBER` and `CONTACT_PARENT` to be the same.

> * You are not allowed to change the name to be the same as any existing student in the list.

Reviewer: ```suggestion
* You are not allowed to change the name of a student to be the same as any existing student in the list.
```

Author: Noted! Thank you.

> * The index **must be a positive integer** 1, 2, 3, …​

Reviewer: ```suggestion
```
This is already included in line 151

Author: Thank you pointing out!

* The index **must be a positive integer** 1, 2, 3, …​

Examples:

> * `list` followed by `delete 2` deletes the 2nd student (if the student exist) in the student list.
> * `find Betsy` followed by `delete 1` deletes the 1st student (if the student exist) in the results of the `find` command.
> * `filter female` followed by `delete 3` deletes the 3rd student (if the student exist) in the results of the `filter` command.

Reviewer: ```suggestion
* `list` followed by `delete 2` deletes the 2nd student (if the student exists) in the student list.
* `find Betsy` followed by `delete 1` deletes the 1st student (if the student exists) in the results of the `find` command.
* `filter female` followed by `delete 3` deletes the 3rd student (if the student exists) in the results of the `filter` command.
```

Author: Edited!

* The index **must be a positive integer** 1, 2, 3, …​
* The given `VALUE_OF_SCORE` can be **any number from 0 to 100**.
* The given `VALUE_OF_SCORE` can either be **an integer or a number with one decimal place**

> * Different exams can have the same `Label`.

Reviewer: ```suggestion
* Exams on different dates can have the same `Label`.
```

Author: Thank you!

* `list` followed by `deletescore 2 1` deletes first score of the 2nd student in the student list.
* `find Betsy` followed by `deletescore 1 2` deletes second score of the 1st student in the results of the `find` command.

### 13. Clearing all entries: `clear`

Clears all entries from the MATHUTORING.

Format: `clear`

### 14. Exiting the program: `exit`

Exits from MATHUTORING.

Format: `exit`

### 15. Filtering students by tag: `filter`

Filters students whose tags match with any of the given keywords.

Format: `filter KEYWORD [MORE_KEYWORDS]`

* The filter is case-insensitive. e.g. `primary` will match `Primary`
* The order of the keywords does not matter. e.g. `primary primary4` will match `primary4 primary`
* Only the tag is searched.
* Only full words will be matched e.g. `primary` will not match `primary4`
* Students whose tag matches at least one keyword will be returned (i.e. `OR` search).
  e.g. `primary primary4` will return `primary`, `primary4`, and `primary primary4`.

Examples:
* `filter primary` returns tag `primary`, `Primary`.
* `filter secondary secondary2` returns tag `secondary`, `secondary2`, and `secondary secondary2`.

Addition:

> * The avatar will change if the student has "gender" tag. 

Reviewer: ```suggestion
* The avatar will change if the student has a "gender" tag. 
```

Author: Changed!

| **Export Data**    | `export [p/FILE_PATH]`<br> e.g. `export p//Users/John`                                                                                                                                        |
| **Export Progress** | `exportp INDEX [p/FILE_PATH]`<br> e.g. `export 1 p//Users/John`                                                                                                                               |
| **Help**           | `help`                                                                                                                                                                                        |
| **Switch**         | `switch`                                                                                                                                                                                      |
| **Exit**           | `exit`                                                                                                                                                                                        |

--------------------------------------------------------------------------------------------------------------------

> ## Appendix Planned Enhancements

Reviewer: Move over to DG

Author: Noted! Thank you so much for checking all the details!
